### DISCUSSION TITLE
How can I log the user out without consent and redirect to homepage?

### DISCUSSION THREAD

**Discussion by l0c0luke** [nextauthjs/next-auth#1005] [Help] [answered]
I don't want the page to show up to ask the user. Just send them straight to the homepage. Is there a way to do this?

**Comment by jaaneh** [comment, marked as answer]
If you're asking for e.g a "log out" button that instantly logs out, you can do something like this.
```js
<div onClick={() => signOut()}>
    Sign out
</div>
```
By calling and invoking the `signOut` method, it will sign the user out on click. Paired with something like this (or your own redirect/auth check method) should work as expected.
```js
export const getServerSideProps = async ({ req }) => {
  const session = await getSession({ req })

  if (!session) {
    return {
      props: {},
      redirect: {
        destination: '/',
        permanent: false
      }
    }
  }

  return {
    props: {}
  }
}
```
